cpp/src/cwrapper/tsfile_cwrapper_expression.cc

```diff
@@ -0,0 +1,207 @@
+#include "writer/tsfile_writer.h"
+
+#define E_OK common::E_OK
+
+#define CONSTRUCT_EXP_INTERNAL(exp, column_name) \
+    do {                                         \
+        exp.column_name = column_name;           \
+        exp.operate_type = oper;                    \
+        exp.children_length = 0;                 \
+    } while (0)
+
+Expression create_column_filter(const char* column_name, OperatorType oper,
+                                int32_t int32_value) {
+    Expression exp;
+    CONSTRUCT_EXP_INTERNAL(exp, column_name);
+    exp.const_condition.value_condition = int32_value;
+    exp.const_condition.type = TSDataType::TS_DATATYPE_INT32;
+    return exp;
+}
+
+Expression create_column_filter(const char* column_name, OperatorType oper,
+                                int64_t int64_value) {
+    Expression exp;
+    CONSTRUCT_EXP_INTERNAL(exp, column_name);
+    exp.const_condition.value_condition = int64_value;
+    exp.const_condition.type = TSDataType::TS_DATATYPE_INT64;
+    return exp;
+}
+Expression create_column_filter(const char* column_name, OperatorType oper,
+                                bool bool_value) {
+    Expression exp;
+    CONSTRUCT_EXP_INTERNAL(exp, column_name);
+    exp.const_condition.value_condition = bool_value ? 1 : 0;
+    exp.const_condition.type = TSDataType::TS_DATATYPE_BOOLEAN;
+    return exp;
+}
+Expression create_column_filter(const char* column_name, OperatorType oper,
+                                float float_value) {
+    Expression exp;
+    CONSTRUCT_EXP_INTERNAL(exp, column_name);
+    memcpy(&exp.const_condition.value_condition, &float_value, sizeof(float));
+    exp.const_condition.type = TSDataType::TS_DATATYPE_FLOAT;
+    return exp;
+}
+Expression create_column_filter(const char* column_name, OperatorType oper,
+                                double double_value) {
+    Expression exp;
+    CONSTRUCT_EXP_INTERNAL(exp, column_name);
+    exp.const_condition.value_condition = double_value;
+    exp.const_condition.type = TSDataType::TS_DATATYPE_DOUBLE;
+    return exp;
+}
+Expression create_column_filter(const char* column_name, OperatorType oper,
+                                const char* char_value) {
+    Expression exp;
+    CONSTRUCT_EXP_INTERNAL(exp, column_name);
+    exp.const_condition.value_condition = reinterpret_cast<int64_t>(char_value);
+    exp.const_condition.type = TSDataType::TS_DATATYPE_TEXT;
+    return exp;
+}
+
+TimeFilterExpression* create_andquery_timefilter() {
+    storage::Expression* exp = new storage::Expression(storage::AND_EXPR);
+    return (TimeFilterExpression*)exp;
+}
+
+TimeFilterExpression* create_time_filter(const char* table_name,
+                                         const char* column_name,
+                                         OperatorType oper, int64_t timestamp) {
+    std::string table_name_str(table_name);
+    std::string column_name_str(column_name);
+    storage::Path path(table_name_str, column_name_str);
```

Comment:
Concatenating the table name with the column name does not result in a correct path because the tags are missing.